File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/savedcards/Card.java
package org.mifos.mobilewallet.core.data.fineract.entity.savedcards;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

/**
 * Created by ankur on 21/May/2018
 */

public class Card implements Parcelable {

    public static final Creator<Card> CREATOR = new Creator<Card>() {
        @Override
        public Card createFromParcel(Parcel source) {
            return new Card(source);
        }

        @Override
        public Card[] newArray(int size) {
            return new Card[size];
        }
    };
    @SerializedName("cardNumber")
    String cardNumber;
    @SerializedName("cvv")
    String cvv;
    @SerializedName("expiryDate")
    String expiryDate;
    @SerializedName("firstName")
    String firstName;
    @SerializedName("lastName")
    String lastName;
    @SerializedName("id")
    int id;

    public Card(String cardNumber, String cvv, String expiryDate, String firstName,
            String lastName) {
        this.cardNumber = cardNumber;
        this.cvv = cvv;
        this.expiryDate = expiryDate;
        this.firstName = firstName;
        this.lastName = lastName;
    }

    protected Card(Parcel in) {
        this.cardNumber = in.readString();
        this.cvv = in.readString();
        this.expiryDate = in.readString();
        this.firstName = in.readString();
        this.lastName = in.readString();
        this.id = in.readInt();
    }

    public String getCardNumber() {
        return cardNumber;
    }

    public void setCardNumber(String cardNumber) {
        this.cardNumber = cardNumber;
    }

    public String getCvv() {
        return cvv;
    }

    public void setCvv(String cvv) {
        this.cvv = cvv;
    }

    public String getExpiryDate() {
        return expiryDate;
    }

    public void setExpiryDate(String expiryDate) {
        this.expiryDate = expiryDate;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.cardNumber);
        dest.writeString(this.cvv);
        dest.writeString(this.expiryDate);
        dest.writeString(this.firstName);
        dest.writeString(this.lastName);
    }

    @Override
    public String toString() {
        return "Card{" +
                "cardNumber='" + cardNumber + '\'' +
                ", cvv='" + cvv + '\'' +
                ", expiryDate='" + expiryDate + '\'' +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", id=" + id +
                '}';
    }
}

File: core/model/src/main/java/com/mifospay/core/model/entity/savedcards/Card.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/mobile-wallet/blob/master/LICENSE.md
 */
package com.mifospay.core.model.entity.savedcards

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class Card(
    @SerializedName("cardNumber")
    var cardNumber: String = " ",

    @SerializedName("cvv")
    var cvv: String = " ",

    @SerializedName("expiryDate")
    var expiryDate: String = " ",

    @SerializedName("firstName")
    var firstName: String = " ",

    @SerializedName("lastName")
    var lastName: String = " ",

    @SerializedName("id")
    var id: Int = 0,
) : Parcelable
